### PR TITLE
argparse migration for cmsDriver and FWCore

### PR DESCRIPTION
**Pull request by kpedro88**
#### PR description:

See #29338. `optparse` has been deprecated for many years. `argparse` offers many functionality improvements, including (relevant to this PR in particular):
* better type enforcement
* automatic handling of exclusive options
* named positional arguments
* automatic formatting of usage strings
* automatic inclusion of default values in help messages
* handling options with different numbers of allowed arguments

While I cannot personally take on migrating every script in CMSSW, at least our most-used common tools should reflect best practices.

In addition, in the course of doing this migration, I fixed various typos and a few stray python3 migration issues.

#### PR validation:

Ran each script to ensure it still works. Unit tests pass.

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Unlikely to be backported.

### CODE REVIEW

File: Configuration/Applications/python/cmsDriverOptions.py [hunk 1]
```diff
@@ -41,21 +41,17 @@ def OptionsFromCommandLine():
...
-            if (str(item) in opt) and (index==len(items)-1 or items[index+1].startswith('-')):
-                items.insert(index+1,value)
-                
-    (options,args) = parser.parse_args(items)
-
-    if not options.conditions or options.conditions=="help":
+    from Configuration.Applications.Options import parser
+
+    options = parser.parse_args(items)
+    print(options)
```
Comment (Dr15Jones) [inline Configuration/Applications/python/cmsDriverOptions.py:47]: was this just for debugging?

Comment (kpedro88) [reply Configuration/Applications/python/cmsDriverOptions.py:47]: yes, let me just rebase that right out of there...

File: FWCore/ParameterSet/test/comparePythonOutput.py [hunk 2]
```diff
@@ -4,15 +4,18 @@
 #  dictionaries or escape characters
 
 from __future__ import print_function
```
Comment (Dr15Jones) [inline FWCore/ParameterSet/test/comparePythonOutput.py:6]: could this line go away?

File: FWCore/PythonUtilities/scripts/compareJSON.py [hunk 3]
```diff
@@ -2,54 +2,55 @@
 
 from __future__ import print_function
```
Comment (Dr15Jones) [inline FWCore/PythonUtilities/scripts/compareJSON.py:3]: obsolete?

File: FWCore/PythonUtilities/scripts/csv2json.py [hunk 4]
```diff
@@ -2,30 +2,27 @@
 
 from __future__ import print_function
```
Comment (Dr15Jones) [inline FWCore/PythonUtilities/scripts/csv2json.py:3]: obsolete?

File: FWCore/PythonUtilities/scripts/filterCSVwithJSON.py [hunk 5]
```diff
@@ -2,34 +2,34 @@
 
 from __future__ import print_function
```
Comment (Dr15Jones) [inline FWCore/PythonUtilities/scripts/filterCSVwithJSON.py:3]: obsolete?

File: FWCore/PythonUtilities/scripts/filterJSON.py [hunk 6]
```diff
@@ -2,27 +2,25 @@
 
 from __future__ import print_function
```
Comment (Dr15Jones) [inline FWCore/PythonUtilities/scripts/filterJSON.py:3]: obsolete?

File: Configuration/Applications/scripts/cmsDriver.py [hunk 7]
```diff
@@ -14,17 +13,6 @@ def run():
         # after cleanup of all config parameters pass it to the ConfigBuilder
         configBuilder = ConfigBuilder(options, with_output = True, with_input = True)
 
-        # Switch on any eras that have been specified. This is not required to create
```
Comment (Dr15Jones) [inline Configuration/Applications/scripts/cmsDriver.py:17]: Where did the era handling go? Or was this not actually needed?

File: FWCore/PythonUtilities/scripts/printJSON.py [hunk 8]
```diff
@@ -2,21 +2,18 @@
 
 from __future__ import print_function
```
Comment (Dr15Jones) [inline FWCore/PythonUtilities/scripts/printJSON.py:3]: obsolete?

File: FWCore/PythonUtilities/scripts/mergeJSON.py [hunk 9]
```diff
@@ -2,11 +2,10 @@
 
 from __future__ import print_function
```
Comment (Dr15Jones) [inline FWCore/PythonUtilities/scripts/mergeJSON.py:3]: obsolete?

File: FWCore/Utilities/scripts/edmCheckClassVersion [hunk 10]
```diff
@@ -1,6 +1,5 @@
 #!  /usr/bin/env python3
 from __future__ import print_function
```
Comment (Dr15Jones) [inline FWCore/Utilities/scripts/edmCheckClassVersion:2]: obsolete?

**Comment by Dr15Jones** [comment]
I wouldn't have commented about the `from __future__` line except you did remove it from one of the files.

File: Configuration/Applications/scripts/cmsDriver.py (hunk 7 above)
Comment (kpedro88) [reply Configuration/Applications/scripts/cmsDriver.py:17]: It was obsoleted by #39127.

**Comment by kpedro88** [comment]
please test

**Comment by kpedro88** [comment]
please test

**Comment by kpedro88** [comment]
Are these relval timeouts becoming more frequent, or am I just unlucky?

**Comment by kpedro88** [comment]
Noticed some more `__future__` imports to remove.

**Comment by kpedro88** [comment]
please test

**Comment by makortel** [comment]
+core

**Comment by antoniovilela** [comment]
+1

**Pull request merged**